lib/streamlit/type_util.py

```diff
@@ -440,6 +445,14 @@ def convert_anything_to_df(df: Any) -> DataFrame:
     if is_type(df, "numpy.ndarray") and len(df.shape) == 0:
         return pd.DataFrame([])
 
+    if is_type(df, _SNOWPARK_DF_TYPE_STR) and not isinstance(df, list):
+        df = pd.DataFrame(df.take(MAX_UNEVALUATED_DF_ROWS))
+        if df.shape[0] == MAX_UNEVALUATED_DF_ROWS:
+            st.caption(
+                f"Showing only 10k rows. Call `collect()` on the dataframe to show more."
```

Comment:
nit: @jrieke should there be a ⚠️ emoji or not?

Reply:
Just talked to @jrieke, and the ⚠️ should be added to the caption as a prefix.

Reply:
@jrieke can confirm, but I'm adding it, because I think is should be there, quoting Johannes I assume the words:

> Change the warning message to

mean it should be clear that it's the warning, thus we need ⚠️, which is a good, informative user experience 

Thank you Lukas for mentioning it out 👍